frontend/src/store/auth/index.js

```diff
@@ -76,16 +108,37 @@ const actions = {
    * Called on application boot once firebase has been initialised
    */
   checkAuth({ commit }) {
+    commit('SET_LOADING', true)
     return new Promise((resolve, reject) => {
       const unsubscribe = auth.onAuthStateChanged(user => {
         unsubscribe()
         resolve(user)
       }, reject)
     })
-      .then(user => commit('SET_USER', user, { root: true }))
+      .then((user) => {
+        if (user === null) throw Error('Not logged in')
+        commit('SET_USER', user, { root: true })
+        return getSelf(user)
+      })
+      .then(profile => {
+        commit('SET_PROFILE', profile, { root: true })
+      })
       .catch(error => {
-        commit('ERROR', error)
+        commit('ERROR', error.message)
+        commit('SET_PROFILE', {}, { root: true })
+        if (!error.code || error.code !== 403) {
```

Comment:
403 is specifically the error we get when there's no profile, but the auth is valid otherwise.
I haven't actually thought about if checkAuth can trigger other errors from our backend, but meh.